src/backend/distributed/utils/statistics_collection.c

```diff
@@ -74,29 +78,46 @@ CollectBasicUsageStatistics(void)
 	StringInfo fields = makeStringInfo();
 	Datum metadataJsonbDatum = 0;
 	char *metadataJsonbStr = NULL;
+	MemoryContext savedContext = CurrentMemoryContext;
 	struct utsname unameData;
+	bool metadataCollectionFailed = false;
 	memset(&unameData, 0, sizeof(unameData));
 
-	StartTransactionCommand();
+	PG_TRY();
+	{
+		distributedTables = DistributedTableList();
```

Comment:
Because we clear OIDs in `InvalidateMetadataSystemCache()`, we end up ignoring regular invalidations, so the metadata cache can be arbitrarily out of date (e.g. `citus_table_size` might always fail after a rebalance).

We have some code for fully invalidating the metadata cache in `InvalidateDistRelationCacheCallback`, which we could wrap in a separate function and call before the call home code.

Reply:
> We have some code for fully invalidating the metadata cache in InvalidateDistRelationCacheCallback, which we could wrap in a separate function and call before the call home code.

Makes sense to me as well. I think it'll also be useful for the future changes on the daemon.

(@pykello pinging you on this comment, in case this is lost among the others. The following code from the mentioned function should do the trick: 

```C
		DistTableCacheEntry *cacheEntry = NULL;
		HASH_SEQ_STATUS status;

		hash_seq_init(&status, DistTableCacheHash);

		while ((cacheEntry = (DistTableCacheEntry *) hash_seq_search(&status)) != NULL)
		{
			cacheEntry->isValid = false;
		}
```

Reply:
Done. Now calling `InvalidateDistRelationCache()` before statistics collection.